src/core/hooks/index.ts

```diff
@@ -1,3 +1,11 @@
+import { AppDispatch, RootState } from 'core/store';
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
+
 export { default as useApiClient } from './useApiClient';
 export { default as useEnv } from './useEnv';
 export { default as useNumericRouteParams } from './useNumericRouteParams';
+
+// Use throughout instead of plain `useDispatch` and `useSelector`
+type DispatchFunc = () => AppDispatch;
+export const useAppDispatch: DispatchFunc = useDispatch;
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
```

Comment:
I would have merged main first, to get these features from there, but now that it's here we can solve it by resolving the conflict that will arise.